scylla/src/transport/timestamp_generator.rs

```diff
@@ -0,0 +1,93 @@
+use std::{
+    ops::DerefMut,
+    sync::atomic::AtomicI64,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+use futures::lock::Mutex;
+use std::sync::atomic::Ordering;
+use tokio::time::{Duration, Instant};
+use tracing::warn;
+
+pub(crate) trait TimestampGenerator {
+    async fn next(&self) -> i64;
+}
```

Comment:
Suggested method's name: `next_timestamp`, for two reasons:
1) not to collide names with possible Iterator's implementation,
2) for explicitness what is returned.